core/tests/ts-integration/tests/custom-account.test.ts

```diff
@@ -192,19 +192,41 @@ describe('Tests for the custom account behavior', () => {
 
         const tx = await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT);
 
-        await expect(
-            sendCustomAccountTransaction(
+        try {
+            await sendCustomAccountTransaction(
                 tx as zksync.types.Transaction,
                 alice.provider,
                 customAccountAddress,
                 testMaster.environment().l2ChainId,
                 undefined,
                 undefined,
                 false
-            )
-        ).toBeRejected(
-            'failed to validate the transaction. reason: Validation revert: Account validation error: Timestamp is out of range'
-        );
+            );
+            expect(null).fail('The transaction was expected to fail');
+        } catch (e) {
+            const err = e as Error;
+            expect(err.message).toContain(
+                'failed to validate the transaction. reason: Validation revert: Account validation error'
+            );
+            const functionSelectorMatch = err.message.match(/function_selector\s=\s(0x[0-9a-fA-F]{8})/);
+            const calldataMatch = err.message.match(/data\s=\s(0x[0-9a-fA-F]+)/);
```

Comment:
Off-topic, probably: Parsing function selector / calldata from the human-readable message looks weird. Is this what Web3 clients expect? I'd imagine an optimal implementation would define separate fields in the JSON-RPC response.

Reply:
Agreed - I'd prefer custom errors to be returned as a part of JSON RPC response.